src/processTargets/modifiers/PositionStage.ts

```diff
@@ -7,66 +7,36 @@ export default class implements ModifierStage {
   constructor(private modifier: PositionModifier) {}
 
   run(context: ProcessedTargetsContext, target: Target): Target[] {
-    const { position } = this.modifier;
     const { start, end } = target.contentRange;
-    const {
-      editor,
-      isReversed,
-      delimiter,
-      scopeType,
-      leadingDelimiter,
-      trailingDelimiter,
-    } = target;
+    const { leadingDelimiter, trailingDelimiter } = target;
 
-    const common = {
-      position: this.modifier.position,
-      editor,
-      isReversed,
-    };
+    target.position = this.modifier.position;
```

Comment:
I think targets should be immutable.  I would prefer to have a function `withPosition(position)` and let it return a new target with the right changes

Reply:
I thought about it but in that case we need the exact same code in every single target. Duplication of code is not something I'm fond of

Reply:
Why? Can't we handle at least part of it in the base class? And if not can't we use composition? Anyways maybe let's just figure it out when we pair today

Reply:
If you wanted to return a new instance of the same derived class that has to be done in each class. But we could do `target.setPosition(position)` then all those changes take place inside the target at least